enh(centreon::common::cps::ups::snmp::mode::batterystatus): enhance cyberpower battery replacement

rmorandell-pgum opened the pull request:
# Community contributors

## Description

Extend the battery-status mode "battery replace indicator", "battery life time", "last replace date" (if exists) and the "time on battery"

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Functionality enhancement or optimization (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## How this pull request can be tested ?

upsAdvanceBatteryReplaceIndicator => replace

<img width="628" height="161" alt="image" src="https://github.com/user-attachments/assets/5d7b66d6-9852-41b4-ac7b-4100aa666aed" />

upsBasicBatteryLastReplaceDate=> last_replace_date

<img width="629" height="166" alt="image" src="https://github.com/user-attachments/assets/1f8b1ce6-a172-4106-acf9-08674967c598" />

upsAdvanceBatteryAgeRecommand=> battery_life_time

<img width="628" height="162" alt="image" src="https://github.com/user-attachments/assets/c5f7d114-c5c7-4663-96e8-2c146e9e7d63" />

Mib

[CyberPower_MIB_v2.11.MIB.txt](https://github.com/user-attachments/files/24125963/CyberPower_MIB_v2.11.MIB.txt)

snmpwalk

[cyberpower.snmpwalk.txt](https://github.com/user-attachments/files/24125966/cyberpower.snmpwalk.txt)


## Checklist

- [x] I have followed the **[coding style guidelines](https://github.com/centreon/centreon-plugins/blob/develop/doc/en/developer/plugins_global.md#5-code-style-guidelines)** provided by Centreon
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (develop).
- [ ] I have provide data or shown output displaying the result of this code in the plugin area concerned.

rmorandell-pgum committed:
extend batterystatus by replace indicator, battery life time, last replace date and time on battery

rmorandell-pgum committed:
Merge branch 'centreon:develop' into cyberpower-battery-replacement

rmorandell-pgum committed:
add %{replace} var to plugin help

omercier commented on src/centreon/common/cps/ups/snmp/mode/batterystatus.pm:
```diff
@@ -144,22 +262,26 @@ Check battery status and charge remaining.
 
 Thresholds.
-Can be: 'charge-remaining' (%), 'voltage' (V), 'temperature' (C).
+Can be: 'charge-remaining' (%), 'voltage' (V), 'temperature' (C), 'timeon' (unit), battery-life-time in months.
```
```suggestion
Can be: C<charge-remaining> (%), C<voltage> (V), C<temperature> (C), C<timeon> (unit), C<battery-life-time> in months.
```

omercier commented on src/centreon/common/cps/ups/snmp/mode/batterystatus.pm:
```diff
@@ -144,22 +262,26 @@ Check battery status and charge remaining.
+
+=item B<--unit>
+
+Select the time unit for thresholds. May be 's' for seconds, 'm' for minutes, 'h' for hours, 'd' for days, 'w' for weeks. Default is seconds.
```
```suggestion
Select the time unit for thresholds. May be 's' for seconds, 'm' for minutes, 'h' for hours, 'd' for days, 'w' for weeks. Default is minutes.
```

rmorandell-pgum committed:
Update src/centreon/common/cps/ups/snmp/mode/batterystatus.pm

Co-authored-by: omercier <32134301+omercier@users.noreply.github.com>

rmorandell-pgum committed:
Update src/centreon/common/cps/ups/snmp/mode/batterystatus.pm

Co-authored-by: omercier <32134301+omercier@users.noreply.github.com>

rmorandell-pgum committed:
Update tests/resources/spellcheck/stopwords.txt

Co-authored-by: omercier <32134301+omercier@users.noreply.github.com>

rmorandell-pgum committed:
Enhacement: convert temperature to undef if empty string to make "no_value => 0"  working on counter.
Fix: remove "no_value => 0" from battery_life_time counter to let be 0 a valid value to raise errors
